docs/design/fabric-v2+/fpc-key-dist.puml

```diff
@@ -180,125 +186,153 @@ group chaincode key distribution
   note left
     if false, abort
   end note
-  Enclave1      -> Enclave1           : check Registered_Enclave2_Credentials matches Enclave2_Credentials
+  Enclave1      -> Enclave1         : extract Enclave2_VK from Enclave2_Credentials
   Enclave1      -> TLCC_Enclave1    ++: can_endorse(Enclave2_VK)
-  note right
-    TLCC checks the endorsement policy to determine
+  note right TLCC_Enclave1
+    Ledger_enclave checks the endorsement policy to determine
     if the recipient enclave is an eligible endorser.
   end note
-  note right
+  note right TLCC_Enclave1
     See above for the what can_endorse does.
-    Contrary to above, here we **do need a secure channel** (as in this context we cannot trust peer)
+    Contrary to above, here we **use a secure channel**
+    (as in this context we cannot trust peer)
   end note
-  return true /'Enclave1 -> TLCC_Enclave1'/
+  return true/false /'Enclave1 -> TLCC_Enclave1'/
   note left
     if false, abort
   end note
-  note over Enclave1
+  note right Enclave1
     Regarding Enclave2's chaincode definition, we can only check that CC_Id matches,
     because the other fields might have changed.
     Alternatively, we could access the "current" chaincode definition,
     and check whether Enclave2's matches it.
   end note
   Enclave1      -> Enclave1         :   check Enclave2_Credentials.Attested_Data.CC_Parameters.CC_Id against internal one
-  Enclave1      -> Enclave1         :   check Enclave2's keys and Enclave2_VK, Enclave2_EK, attestation_report
-  Enclave1      -> Enclave1         :   encrypted_cckeys_message <- encrypt <SEK, Chaincode_DK> with Enclave2_EK
+  Enclave1      -> Enclave1         :   check Enclave2_VK, Enclave2_EK, ATTESTATION
```

Comment:
(follow up on a comment of Michael's)
```suggestion
  note right Enclave1
    The attestation verification requires to have the IAS keys inside the enclave.
    Alternatively, this check can be removed, thus relying on ERCC and TLCC.
    ERCC already verified the attestation as part of the enclave registration,
    using the IAS keys which are part of ERCC itself.
    Also, TLCC previously confirmed that such registration was committed.
  end note
  Enclave1      -> Enclave1         :   check Enclave2_VK, Enclave2_EK, ATTESTATION
```

Reply:
didn't notice that attestation was still there. good point to add that